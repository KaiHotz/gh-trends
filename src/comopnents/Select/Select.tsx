import React, { FC, useState, SyntheticEvent, ReactNode } from 'react';
import { SelectMenu } from '@primer/components';
import { StyledButtonInvisible } from '../StyledComponents';

export interface ISelectOption {
  value?: string;
  label: string;
}

export interface ISelectProps {
  options: ISelectOption[];
  label: ReactNode;
  filterTxt: string;
  value?: string;
  onChange: (val?: string) => void;
}

export const Select: FC<ISelectProps> = ({ options, label, filterTxt, value, onChange }) => {
  const [filter, setFilter] = useState('');

  const filteredOptions = options.filter((option) => option.label.toLowerCase().startsWith(filter.toLowerCase()));

  const handleChange = (val?: string) => onChange(val);

  return (
    <SelectMenu css={{}} defaultValue="foo">
      <StyledButtonInvisible as="summary">{label}</StyledButtonInvisible>
      <SelectMenu.Modal>
        <SelectMenu.Filter
          placeholder={filterTxt}
          value={filter}
          aria-label={filterTxt}
          onChange={(e: SyntheticEvent<HTMLInputElement>) => setFilter(e.currentTarget.value)}
        />
        <SelectMenu.List>
          {filteredOptions.map((option) => (
            <SelectMenu.Item key={option.value} selected={value === option.value} onClick={() => handleChange(option.value)}>
              {option.label}
            </SelectMenu.Item>
          ))}
        </SelectMenu.List>
      </SelectMenu.Modal>
    </SelectMenu>
  );
};
