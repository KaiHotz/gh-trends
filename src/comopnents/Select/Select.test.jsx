import React from 'react';
import { shallow } from 'enzyme';
import { Select } from './Select';

const defaultProps = {
  options: [
    { value: 'foo', label: 'Foo' },
    { value: 'bar', label: 'Bar' },
  ],
  label: 'this is a label',
  filterTxt: 'This is the Filter text',
  value: 'foo',
  onChange: jest.fn(),
};

describe('Select', () => {
  it('should render', () => {
    const wrapper = shallow(<Select {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
