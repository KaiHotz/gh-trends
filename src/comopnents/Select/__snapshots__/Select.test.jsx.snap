// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render 1`] = `ShallowWrapper {}`;
